### PR TITLE
feat(alerts): add spike protection notification setting

### PR DESCRIPTION
**Pull request by cathteng**
Backend part for spike protection notification settings.

Eventually should look something like this
Screen Shot 2022-10-26 at 3 41 11 PM

### CODE REVIEW

File: src/sentry/models/notificationsetting.py [hunk 1]
```diff
@@ -81,6 +81,8 @@ def provider_str(self) -> str:
             (NotificationSettingTypes.QUOTA_TRANSACTIONS, "quotaTransactions"),
             (NotificationSettingTypes.QUOTA_ATTACHMENTS, "quotaAttacments"),
             (NotificationSettingTypes.QUOTA_WARNINGS, "quotaWarnings"),
+            (NotificationSettingTypes.QUOTA_SPEND_ALLOCATIONS, "quotaSpendAllocations"),
+            (NotificationSettingTypes.SPIKE_PROTECTION, "spikeProtection"),
```
Comment (cathyteng17) [inline src/sentry/models/notificationsetting.py:85]: also added this change because i didn't do this for spend allocation alerts

**Review by scefali — COMMENTED** [review]
@cathyteng17 there's no way to fine-tune these settings on a per-project or per-organization level? If so, we should make sure to link to that UI

**Comment by cathyteng17** [comment]
> @cathyteng17 there's no way to fine-tune these settings on a per-project or per-organization level? If so, we should make sure to link to that UI

@scefali this will probably be something for later? in that case maybe we could keep the fine tune but just have nothing on the page it links to?

**Comment by scefali** [comment]
@cathyteng17 
Well, I'd imagine it would automatically be fine-tunable if we set up the configuration correctly. It should just work out of the box.

**Comment by cathyteng17** [comment]
@scefali i updated the screenshot, but this PR is just the backend part for now. please let me know if i'm missing anything if i want it to work out of the box

**Review by AniketDas-Tekky — COMMENTED** [review]
Can you add a unit test to verify defaults?

**Comment by scefali** [comment]
@AniketDas-Tekky FWIW I don't think we need unit tests for this PR. It's largely just config changes. Better to have good integration-level tests for the actual notification making sure we can read from the settings.

**Review by scefali — APPROVED** [review]

**Pull request merged**